src/main/java/com/epam/onlinestore/repository/impl/ProductRepositoryImpl.java

```diff
@@ -0,0 +1,48 @@
+package com.epam.onlinestore.repository.impl;
+
+import com.epam.onlinestore.entity.Product;
+import com.epam.onlinestore.repository.ProductRepository;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * a Layer class that contains methods for working with a database with a Product
+ */
+@Slf4j
+@Component
```

Comment:
Use @Repository annotation instead of @Component, it has actually no difference, but first one has more readability to reviewers. Check other repositories for the same issue